lib/jose-jwe-decrypt.js

```diff
@@ -21,35 +21,51 @@
  * @param plain_text   string to decrypt
  * @return Promise<string>
  */
-JoseJWE.prototype.decrypt = function(key_promise, cipher_text) {
+JoseJWE.decrypt = function(cryptographer, key_promise, cipher_text) {
+  /**
+   * Decrypts the encrypted CEK. If decryption fails, we create a random CEK.
+   *
+   * In some modes (e.g. RSA-PKCS1v1.5), you myst take precautions to prevent
+   * chosen-ciphertext attacks as described in RFC 3218, "Preventing
+   * the Million Message Attack on Cryptographic Message Syntax". We currently
+   * only support RSA-OAEP, so we don't generate a key if unwrapping fails.
+   *
+   * return Promise<CryptoKey>
+   */
+  var decryptCek = function(key_promise, encrypted_cek) {
+    return key_promise.then(function(key) {
+      return cryptographer.unwrapCek(encrypted_cek, key);
+    });
+  };
+
   // Split cipher_text in 5 parts
   var parts = cipher_text.split(".");
   if (parts.length != 5) {
-    return Promise.reject("decrypt: invalid input");
+    return Promise.reject(Error("decrypt: invalid input"));
   }
 
   // part 1: header
   header = JSON.parse(Utils.Base64Url.decode(parts[0]));
   if (!header.alg) {
-    return Promise.reject("decrypt: missing alg");
+    return Promise.reject(Error("decrypt: missing alg"));
   }
-  this.setKeyEncryptionAlgorithm(header.alg);
+  cryptographer.setKeyEncryptionAlgorithm(header.alg);
```

Comment:
We could prevent the mutation with:

    var new_cryptographer = new cryptographer.constructor;